### PR TITLE
Add and Implement Theme Colors

### LINKED ISSUE #777: [nit] Table elements misaligned in Application UI
**Describe the bug**
The message columns in the tables for Source and Automation conditions are out of line and would be nicer if they matched. 

**Screenshots**
nit - wgo UI bug

### LINKED ISSUE #914: UI: remove user icon whilst no account menu exists
UI banner currently includes an icon for a person as a placeholder for when we have account information/preferences. Whilst this doesnt exist, it is confusing to show this as appears should be a clickable button.

### LINKED ISSUE #1096: UI cleanup sizes and colors
A list of nits, with reference to our new [design system colors](https://www.figma.com/file/gnodK9xtoOahXBqa6ONsA5/Design-System-%2F-Draft?node-id=4%3A6) : 

- [ ] All dark/black text should be `#1A1A1A`, with no transparencies

Screenshot 2021-11-18 at 13 43 19

- [ ] All grey text should be `#737373`, with no transparencies
Screenshot 2021-11-18 at 14 31 31

- [ ] grey and dark icons should also be `#737373` and `#1A1A1A`
Screenshot 2021-11-18 at 14 43 33

- [ ] Buttons in color `#00A7CC` should be `#006B8E` ([dark primary color](https://www.figma.com/file/gnodK9xtoOahXBqa6ONsA5/Design-System-%2F-Draft?node-id=4%3A6)) with white semibold capitalised text
Screenshot 2021-11-18 at 14 02 01

- [ ] Provider dropdown should have a bold label and table width should be equal to tile page width in order to have a smooth transition
chrome-capture

- [ ] This random pink slider should be blue
Screenshot 2021-11-18 at 13 53 31

- [ ] Menu item marker should be `#00B3EC` instead of `#00d2ff`
Screenshot 2021-11-18 at 14 27 34

### PR DESCRIPTION
**Pull request by joshri**
Parts of #1096 are completed in this PR - some parts are exclusive to Enterprise and are not fixed here. 
Closes #777 
Closes #914 

Theme colors - adds primary and text keys to customized mui palette, fixing black and grey text, light blue buttons, and AddApplication switch color

Aligns tables in ApplicationDetail with widths array prop

Removes Account Icon until functionality is added

### CODE REVIEW

File: ui/components/ConditionsTable.tsx [hunk 1]
```diff
@@ -18,31 +10,23 @@ type Props = {
...
-                <TableCell>{c.message}</TableCell>
-              </TableRow>
-            ))}
-          </TableBody>
-        </Table>
-      </TableContainer>
-    </div>
+    <DataTable
+      fields={[
+        { label: "Type", value: (c) => c.type },
```
Comment (jpellizzari) [inline ui/components/ConditionsTable.tsx:15]: If I remember correctly how the API for this work, this should simplify things:
```suggestion
        { label: "Type", value: 'type' },
```

File: ui/lib/theme.ts [hunk 2]
```diff
@@ -97,8 +98,16 @@ export const GlobalStyle = createGlobalStyle`
 export const muiTheme = createTheme({
   typography: { fontFamily: "proxima-nova" },
   palette: {
+    primary: {
+      main: "#006B8E",
```
Comment (jpellizzari) [inline ui/lib/theme.ts:102]: Nit: comment about what this color is, like "Weave Blue" or something.

File: ui/components/ConditionsTable.tsx (hunk 1 above)
Comment (joshri) [reply ui/components/ConditionsTable.tsx:15]: YES

File: ui/lib/theme.ts (hunk 2 above)
Comment (joshri) [reply ui/lib/theme.ts:102]: YES

File: ui/components/ConditionsTable.tsx (hunk 1 above)
Comment (jpellizzari) [reply ui/components/ConditionsTable.tsx:15]: > Wait - isn't it getting the type from the Condition object?

Yes. If the table finds a function, it runs it; else it returns the value. This allows you to get the value for a column from somewhere else on the object if you need to, but also makes the default case easy.

Comment (joshri) [reply ui/components/ConditionsTable.tsx:15]: > 

Wow COOL. fixed.

**Review by jpellizzari — APPROVED** [review]
Looks good :+1:

**Pull request merged**